tools/flash-shelly.py

```diff
@@ -0,0 +1,664 @@
+      perform_flash = True
+      if mode == 'stock':
+        keyword = "converted to Official firmware"
+      elif mode == 'homekit':
+        keyword = "converted to HomeKit firmware"
+    elif current_fw_type == mode and isNewer(flash_fw_version, current_fw_version):
+      perform_flash = True
+      keyword = f"upgraded from {current_fw_version} to version {flash_fw_version}"
+    elif not dlurl:
+      if version:
+        keyword = f"Version {version} is not available..."
+      else:
+        keyword = "Is not supported yet..."
+      logger.info(f"{keyword}\n")
+      return 0
+    else:
+      logger.info("Does not need flashing...\n")
+      return 0
+
+    logger.debug(f"\nperform_flash: {perform_flash}\n")
+    if perform_flash == True and dry_run == False and silent_run == False:
+      if requires_upgrade == True:
+        flash_message = f"{message} {keyword}"
+      else:
+        flash_message = f"Do you wish to flash {friendly_host} to firmware version {flash_fw_version}"
+      if input(f"{flash_message} (y/n) ? ") == 'y':
+        flash = True
+      else:
+        flash = False
+        logger.info("Skipping Flash...")
+    elif perform_flash == True and dry_run == False and silent_run == True:
+      flash = True
+    elif perform_flash == True and dry_run == True:
+      logger.info(f"{message} {keyword}...")
+    if flash == True:
+      write_flash(device_info, hap_setup_code)
+  logger.info(" ")
+
+
+def device_scan(hosts, action, do_all, dry_run, silent_run, mode, exclude, version, variant, hap_setup_code):
+  logger.debug(f"\n{WHITE}device_scan{NC}")
+  requires_upgrade = False
+
+  if not do_all:
+    device_list = []
+    logger.info(f"{WHITE}Probing Shelly device for info...\n{NC}")
+    for host in hosts:
+      deviceinfo = Device(host)
+      deviceinfo.getDeviceInfo()
+      if deviceinfo.fw_type is not None:
+        dict = {'host': deviceinfo.host, 'wifi_ip': deviceinfo.wifi_ip, 'fw_type': deviceinfo.fw_type, 'device_url': deviceinfo.device_url, 'info': deviceinfo.info}
+        device_list.append(dict)
+  else:
+    logger.info(f"{WHITE}Scanning for Shelly devices...{NC}")
+    zc = zeroconf.Zeroconf()
+    listener = MyListener()
+    browser = zeroconf.ServiceBrowser(zc, '_http._tcp.local.', listener)
+    count = 1
+    total_loop = 1
+    while count < 10:
+      nod = len(listener.device_list)
+      time.sleep(2)
+      count += 1
+      if len(listener.device_list) == nod:
+        total_loop += 1
+      if total_loop > 3:
+        break
+    zc.close()
+    device_list = listener.device_list
+    nod = len(device_list)
+    logger.info(f"{GREEN}{nod} Devices found.\n{NC}")
+  sorted_device_list = sorted(device_list, key=lambda k: k['host'])
+  logger.trace(f"device_test: {sorted_device_list}")
+
+  for device in sorted_device_list:
+    if mode == 'keep':
+      flashmode = device['fw_type']
+    else:
+      flashmode = mode
+
+    if device['fw_type'] == 'homekit':
```

Comment:
nice, i like this part, much better than it used to be!

Reply:
ok done changed to `lower_case_underscore_style()`